packages/dashboard-core-plugins/src/ChartPlugin.tsx

```diff
@@ -37,9 +37,10 @@ export function ChartPlugin(props: ChartPluginProps): JSX.Element | null {
       }
 
       const metadata = { name, figure: name };
+      const makeApi = () => dh;
```

Comment:
Instead of a `makeApi`, we should use contexts, and just `useApi` from here, e.g.
```
const dh = useApi();
```

Also just wrapping the child panels with the context provider as well. #1229 will make that simpler.

Also - `dh` seems to be the global instance here. I'm not sure why it's not causing a compilation error? Perhaps remove the dependency on `jsapi-shim` in `dashboard-core-plugins` and then see...

Reply:
I'm splitting the changes into separate PRs, this PR de-globalizes the API in the Chart package with minimal changes in the consumers.
Are we planning to land #1229 in vermilion?

Reply:
Disregard that comment about #1229, it won't be relevant for this anyway, since it'll be on a per panel basis, and we need to load the API asynchronously. This method is fine.